File: Pattern.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public class Pattern {
    private static int readSize(String fileName) throws IOException {
        String text = Files.readString(Path.of(fileName)).trim();
        if (text.contains("=")) {
            text = text.split("=", 2)[1].trim();
        }

        int size = Integer.parseInt(text);
        if (size <= 0) {
            throw new IllegalArgumentException("pattern size must be positive");
        }
        return size;
    }

    private static String makePattern(int size) {
        StringBuilder pattern = new StringBuilder();
        for (int row = 1; row <= size; row++) {
            pattern.append(" ".repeat(size - row));
            pattern.append("*".repeat(2 * row - 1));
            if (row < size) {
                pattern.append(System.lineSeparator());
            }
        }
        return pattern.toString();
    }

    public static void main(String[] args) throws IOException {
        if (args.length == 0) {
            System.out.println("Usage: java Pattern <parameter-file> [parameter-file ...]");
            return;
        }

        for (int index = 0; index < args.length; index++) {
            int size = readSize(args[index]);
            if (index > 0) {
                System.out.println();
            }
            System.out.println(args[index] + " -> size " + size);
            System.out.println(makePattern(size));
        }
    }
}

File: pattern.py
#!/usr/bin/env python3
"""Print star patterns from parameter files."""

from pathlib import Path
import sys


def read_size(file_name: str) -> int:
    text = Path(file_name).read_text(encoding="utf-8").strip()
    if "=" in text:
        text = text.split("=", 1)[1].strip()

    size = int(text)
    if size <= 0:
        raise ValueError("pattern size must be positive")
    return size


def make_pattern(size: int) -> str:
    rows = []
    for row in range(1, size + 1):
        spaces = " " * (size - row)
        stars = "*" * (2 * row - 1)
        rows.append(spaces + stars)
    return "\n".join(rows)


def main() -> int:
    if len(sys.argv) < 2:
        print("Usage: python3 pattern.py <parameter-file> [parameter-file ...]")
        return 1

    for index, file_name in enumerate(sys.argv[1:]):
        size = read_size(file_name)
        if index:
            print()
        print(f"{file_name} -> size {size}")
        print(make_pattern(size))

    return 0


if __name__ == "__main__":
    raise SystemExit(main())
